docs/examples/kotest/src/test/kotlin/com/example/kotest/TestContainerKotestIntegrationPerTest.kt

```diff
@@ -0,0 +1,39 @@
+package com.example.kotest
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.delay
+import org.testcontainers.containers.ContainerPerTestListener
+import org.testcontainers.containers.GenericContainer
+
+class TestContainerKotestIntegrationPerTest : StringSpec({
+    val container = GenericContainer<Nothing>("ubuntu")
+    setContainerCommand(container) // Set container command for first time to echo hello Kotest
+
+    // Register the Listener
+    listeners(ContainerPerTestListener(container))
+
+    "test should echo hello Kotest for first test" {
+        container.logs.trim() shouldBe "hello Kotest"
+        // Set container command so that next time when container start we will have hello Kotest again in container log
+        setContainerCommand(container)
+        delay(1000)
```

Comment:
Why `delay`?

Reply:
Without delay i was getting following error for the second test during container startup

```bash
09:40:15.944 [ducttape-0] DEBUG com.github.dockerjava.core.command.AbstrDockerCmd - Cmd: cb26186a3bb004fcd296bedcbb93b3ecdd93e18d69fba2559242a3933de219a9,false
09:40:15.945 [ducttape-0] DEBUG com.github.dockerjava.core.exec.InspectContainerCmdExec - GET: com.github.dockerjava.okhttp.OkHttpWebTarget@7eef3f6f
09:40:16.953 [ducttape-0] DEBUG com.github.dockerjava.core.command.AbstrDockerCmd - Cmd: cb26186a3bb004fcd296bedcbb93b3ecdd93e18d69fba2559242a3933de219a9,false
09:40:16.953 [ducttape-0] DEBUG com.github.dockerjava.core.exec.InspectContainerCmdExec - GET: com.github.dockerjava.okhttp.OkHttpWebTarget@22be88f0
09:40:17.964 [ducttape-0] DEBUG com.github.dockerjava.core.command.AbstrDockerCmd - Cmd: cb26186a3bb004fcd296bedcbb93b3ecdd93e18d69fba2559242a3933de219a9,false
09:40:17.964 [ducttape-0] DEBUG com.github.dockerjava.core.exec.InspectContainerCmdExec - GET: com.github.dockerjava.okhttp.OkHttpWebTarget@3ae5ee78

```
With a very long list of this warning, eventually test  was failing with 

```
org.testcontainers.containers.ContainerLaunchException: Container startup failed

	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:320)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:300)
	at org.testcontainers.containers.StartablePerTestListener.beforeTest(StartablePerTestListener.kt:22)
	at io.kotest.core.runtime.CallbacksKt.invokeBeforeTest(callbacks.kt:40)
	at io.kotest.core.runtime.TestExecutor.executeActiveTest(TestExecutor.kt:116)
	at io.kotest.core.runtime.TestExecutor$intercept$2.invokeSuspend(TestExecutor.kt:74)
	at io.kotest.core.runtime.TestExecutor$intercept$2.invoke(TestExecutor.kt)
	at io.kotest.core.runtime.TestExecutor.executeIfActive(TestExecutor.kt:86)
	at io.kotest.core.runtime.TestExecutor.intercept(TestExecutor.kt:74)
	at io.kotest.core.runtime.TestExecutor.execute(TestExecutor.kt:55)
	at io.kotest.core.engine.SingleInstanceSpecRunner.runTest(SingleInstanceSpecRunner.kt:62)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$2.invokeSuspend(SingleInstanceSpecRunner.kt:73)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$2.invoke(SingleInstanceSpecRunner.kt)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$3.invokeSuspend(SingleInstanceSpecRunner.kt:79)
	at io.kotest.core.engine.SingleInstanceSpecRunner$execute$3.invoke(SingleInstanceSpecRunner.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:91)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:177)
	at io.kotest.core.engine.SingleInstanceSpecRunner.execute(SingleInstanceSpecRunner.kt:77)
	at io.kotest.core.engine.SpecExecutor$runTests$run$1.invokeSuspend(SpecExecutor.kt:105)
	at io.kotest.core.engine.SpecExecutor$runTests$run$1.invoke(SpecExecutor.kt)
	at io.kotest.core.engine.SpecExecutor.interceptSpec(SpecExecutor.kt:117)
	at io.kotest.core.engine.SpecExecutor.runTests(SpecExecutor.kt:108)
	at io.kotest.core.engine.SpecExecutor.execute(SpecExecutor.kt:36)
	at io.kotest.core.engine.KotestEngine$submitBatch$$inlined$forEach$lambda$1$1.invokeSuspend(KotestEngine.kt:66)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:272)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:79)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:54)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:36)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at io.kotest.core.engine.KotestEngine$submitBatch$$inlined$forEach$lambda$1.run(KotestEngine.kt:65)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:313)
	... 37 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:488)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:315)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	... 38 more
Caused by: org.rnorth.ducttape.TimeoutException: Timeout waiting for result with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:54)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilTrue(Unreliables.java:100)
	at org.testcontainers.containers.startupcheck.StartupCheckStrategy.waitUntilStartupSuccessful(StartupCheckStrategy.java:35)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:423)
	... 40 more
Caused by: java.lang.RuntimeException: Not ready yet
	at org.rnorth.ducttape.unreliables.Unreliables.lambda$retryUntilTrue$1(Unreliables.java:102)
	at org.rnorth.ducttape.unreliables.Unreliables.lambda$retryUntilSuccess$0(Unreliables.java:43)
	... 4 more

```

Reply:
@bsideup  Is there any other ways in testcontainers to avoid this error without using explicit delay?

Reply:
try `.withStartupCheckStrategy(new OneShotStartupCheckStrategy())`